### ISSUE TITLE
Mismatch on hugginface between model config and tokenizer config ?

### ISSUE THREAD

**Issue by LaurentVidal95** [HUBioDataLab/SELFormer#19]
Dear SELFormer devs, 

First thank you for your very nice paper! I wanted to finetune SELFormer for a personal work. 

But there seems to be a mismatch on the hugging face repo between the `config.json` and the `tokenizer_config.json`.
I'm not sure I'm supposed to raise the issue here, but the last dicussion on HF is two years old.

I simply called both the model and tokenizer from huggin face with :
```python
from transformers import AutoModelForCausalLM, AutoTokenizer

model = AutoModelForCausalLM.from_pretrained("HUBioDataLab/SELFormer")
tokenizer = AutoTokenizer.from_pretrained("HUBioDataLab/SELFormer")

print("Tokenizer length:", len(tokenizer))
print("Model vocab size:", model.get_input_embeddings().weight.size(0))

print("Tokenizer pad_token_id:", tokenizer.pad_token_id)
print("Model config pad_token_id:", model.config.pad_token_id)
print("Embedding padding_idx:", model.get_input_embeddings().padding_idx)
```

with the following mismatch:
```sh
Tokenizer length: 428
Model vocab size: 800
Tokenizer pad_token_id: 3
Model config pad_token_id: 1
Embedding padding_idx: 1
```

I'm not sure that's intented. Is this normal ?

**Comment by ervau** [comment]
Hi, thanks for reaching out and for your interest in our model. We've looked into this, and you're right about the mismatches. Here’s a detailed breakdown of why they exist and why they aren't critical for model performance.

**- Vocab size mismatch:** This is intentional and completely harmless. The model's embedding matrix was initialized with a vocab_size of 800 during pre-training, serving as a buffer for potential future tokens. However, the final tokenizer, trained on the SELFIES dataset, only contains 428 unique tokens. The unused embedding rows remain inactive and do not affect the model's behavior or performance.

**- Pad token ID mismatch:** This is a small configuration artifact rather than a weight-level problem. The model's functionality relies on the attention_mask generated by the tokenizer, not the pad_token_id in its config. The tokenizer correctly uses ID 3 for padding and produces an attention_mask that ensures these padded positions are ignored during computation. Therefore, this mismatch does not negatively impact inference or fine-tuning.

While the model works correctly as is, you can align the configuration before fine-tuning or further training to keep everything consistent:
```python
model.config.pad_token_id = tokenizer.pad_token_id
model.get_input_embeddings().padding_idx = tokenizer.pad_token_id
model.resize_token_embeddings(len(tokenizer)
```

**Comment by LaurentVidal95** [comment]
Thank you very much for your reply that's clearer now ! I'm closing the issue then 👍
